File: c++/6 kyu Playing with digits.cpp
#include<math.h>
class DigPow
{
public:
  static int digPow(int n, int p){
  int sum=0,m=floor(log10(n)+1),a=n;
	m+=p-1;
	while(n>0){
  	sum+=pow((n%10),m--);
	n=n/10;	}
	if(sum%a==0)return sum/a;
	return -1;
  }
};

File: Kotlin/6 kyu Playing with digits.kt
fun digPow(n: Int, p: Int): Int {
 var s:String=n.toString();
	var a=p;
	var sum:Int = 0;
	for(i in s)
		sum+= Math.pow( i.toDouble()-'0'.toDouble(),a++.toDouble()).toInt();
	if(sum%n==0)return sum/n;
	return -1;
}
